### ISSUE TITLE
deploy broken: mime-types-data requires Ruby version >= 2.0.

### ISSUE THREAD

**Issue by fasterthanlime** [travis-ci/travis-ci#5145]
Looks like the deploy step is broken because [mime-types 3.0 requires ruby 2.0](https://rubygems.org/gems/mime-types)

[Example log](https://travis-ci.org/itchio/elevate/jobs/92455729#L216)

Could you either:

  - Upgrade the deploy step to use ruby 2.0+ (sounds risky though)
  - Specify a more restrictive dependency on mime-types, `~>2.0` for example

Thanks in advance! :star2:

**Comment by mnutt** [comment]
We ran into this and it was keeping us from deploying. A workaround that worked for us:

```
before_deploy:
- gem install mime-types -v 2.6.2
```

**Comment by fasterthanlime** [comment]
My workaround was:

```
deploy:
  edge:
    source: fasterthanlime/dpl
    branch: patch-1
  # normal deploy params
```

but yours is much sweeter :)

I have a PR open at https://github.com/travis-ci/dpl/pull/365 but I'm not sure how good of a fix it is.

**Comment by ruslo** [comment]
> A workaround that worked for us

This fix [linux][1] for me but not [osx][2].

[1]: https://travis-ci.org/forexample/github-binary-release/jobs/92486571
[2]: https://travis-ci.org/forexample/github-binary-release/jobs/92486575

**Comment by zhengshaolin** [comment]
@mnutt awesome fix man!

**Comment by oryband** [comment]
this still happens to me on a linux private build when uploading to s3:
```
Fetching: dpl-1.8.8.gem (100%)
Successfully installed dpl-1.8.8
1 gem installed

Installing deploy dependencies
Fetching: jmespath-1.1.3.gem (100%)
Successfully installed jmespath-1.1.3
Fetching: aws-sdk-core-2.2.1.gem (100%)
Successfully installed aws-sdk-core-2.2.1
Fetching: aws-sdk-resources-2.2.1.gem (100%)
Successfully installed aws-sdk-resources-2.2.1
Fetching: aws-sdk-2.2.1.gem (100%)
Successfully installed aws-sdk-2.2.1
4 gems installed
Fetching: mime-types-data-3.2015.1120.gem (100%)
ERROR:  Error installing mime-types:
    mime-types-data requires Ruby version >= 2.0.
/home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- mime-types (LoadError)
    from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider.rb:76:in `requires'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider/s3.rb:7:in `<class:S3>'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider/s3.rb:5:in `<class:Provider>'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider/s3.rb:4:in `<module:DPL>'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider/s3.rb:3:in `<top (required)>'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider.rb:58:in `const_get'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider.rb:58:in `block in new'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/cli.rb:41:in `fold'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/provider.rb:55:in `new'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/cli.rb:31:in `run'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/lib/dpl/cli.rb:7:in `run'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.8/bin/dpl:5:in `<top (required)>'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `load'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `<main>'
failed to deploy
```

**Comment by oryband** [comment]
@avivl @fuzzyami @doodyparizada @liorama @gadisr

**Comment by oryband** [comment]
we can't deploy to production because of this. aws s3 deploys are broken.

**Comment by fasterthanlime** [comment]
@oryband yup, the issue is still open, waiting on a response from the @travis-ci team — in the meantime, mnutt's workaround will probably work for you! https://github.com/travis-ci/travis-ci/issues/5145#issuecomment-158677487 :nut_and_bolt:

**Comment by oryband** [comment]
@fasterthanlime thanks, using this hack in the meanwhile.

**Comment by sreimers** [comment]
to fix this on linux and osx you can try this modified workaround:

```
before_deploy:
- rvm 1.9.3 do gem install mime-types -v 2.6.2
```

https://travis-ci.org/Studio-Link-v2/backend/jobs/92562905

**Comment by ruslo** [comment]
@sreimers It works, thanks!

**Comment by jwvanderbeck** [comment]
Having this issue as well

**Comment by BanzaiMan** [comment]
Sorry for the troubles, everyone. We've just pushed dpl 1.8.9 with the fix by @fasterthanlime. Let us know if you still see issues with this version.

**Comment by oryband** [comment]
i can confirm this our builds are again working.

**Comment by ruslo** [comment]
Still failed, even with `dpl 1.8.9`:
```
Fetching: dpl-1.8.9.gem (100%)
```
* Linux https://travis-ci.org/forexample/github-binary-release/jobs/92706898
* OSX https://travis-ci.org/forexample/github-binary-release/jobs/92706905

**Comment by oryband** [comment]
@BanzaiMan as a side-question, why doesn't travis wrap these deploy dependency tools in some sort of docker container, and save us from having to install these on every deploy build? i don't care which dependencies and tools you're using (ruby stuffin this case), i just want a binary to work. doing this in docker could avoid having to install these deps every time.

**Comment by BanzaiMan** [comment]
@ruslo Thanks for the report. I missed one. v1.8.10 will be out shortly.

**Comment by BanzaiMan** [comment]
@oryband Docker is not available on all build images.

**Comment by kushdilip** [comment]
Builds are working fine now.

**Comment by oryband** [comment]
@BanzaiMan why not?

**Comment by hdgarrood** [comment]
I was experiencing this too and `dpl 1.8.10` has fixed the problem for me. Thanks! :smile:

**Comment by BanzaiMan** [comment]
@oryband Thanks for the input, but it is not feasible at this time.

**Comment by dgmstuart** [comment]
Hi - I'm not sure I understand the fix: I'm using https://travis-ci.org and builds are still failing. The workarounds don't result in successful builds:

https://github.com/sgruhier/foundation_rails_helper/pull/114
https://github.com/sgruhier/foundation_rails_helper/pull/116

Is there somewhere I can set some config to specify the dpl version to be used?

**Comment by dgmstuart** [comment]
Nevermind - ours is a separate issue - unrelated to Travis

**Comment by refeuerlein** [comment]
Still having this problem. Opened last year. ?????????????

**Comment by BanzaiMan** [comment]
@refeuerlein Sorry for the troubles. Sounds like something else updated to require a newer version of `mime-types`. Please open a separate issue, indicating which provider you are using, and pointing to a build log URL. Thanks.

**Comment by saikrishnagaddipati** [comment]
Same issue

**Comment by mminski** [comment]
Fixed problem by downgrading from rails '4.2.6' to rails '4.1.0'.

**Comment by ncri** [comment]
Just noticed my nested_form_fields gem seems to have this issue:

https://travis-ci.org/ncri/nested_form_fields

Anyone any idea what I could do to get the 1.9.3 build pass?

**Comment by BanzaiMan** [comment]
@ncri Looking at https://rubygems.org/gems/public_suffix/versions, https://rubygems.org/gems/public_suffix/versions/1.4.6 was the last version of `public_suffix` that run on 1.9.3. Please do note that your issue is completely separate from the issue described here.

**Comment by ncri** [comment]
Oh, sorry for this, not sure how I ended up here. Is there an issue open for this? Or should I open one? public_suffix is not needed by my gem, travis seems to add it. At least it's not in my Gemfile.lock. Or is it possible that the gemspec had a gem depending on? But wouldn't it be written into the .lock file in that case? 

So it seems I cannot do anything about it myself.

**Comment by ncri** [comment]
Ah, never mind, the Gemfile.lock file is not checked in anyway, so I assume I need to enforce a version in one of the gems in my gemspec. Hard to find out which one has the public_suffix dependency though. Any pointers? Thanks! I'm out here then ;-)

**Comment by BanzaiMan** [comment]
`gem dep` will show you.

**Comment by ncri** [comment]
thx
